File: node-api-pro/kalkulasi.js
// Fungsi mencari rata-rata
function hitungRataRata(data) {
    const total = data.reduce((sum, item) => sum + item.nilai, 0);
    return (total / data.length).toFixed(2);
}

// Fungsi menghitung jumlah status
function rekapStatus(data) {
    const rekap = { Normal: 0, Warning: 0, Critical: 0 };
    data.forEach(item => {
        if (rekap[item.status] !== undefined) {
            rekap[item.status]++;
        }
    });
    return rekap;
}

module.exports = { hitungRataRata, rekapStatus };

File: python-api-pro/kalkulasi.py
def hitung_rata_rata(data):
    if not data:
        return 0
    total = sum(item["nilai"] for item in data)
    return round(total / len(data), 2)

def rekap_status(data):
    # Siapkan wadah untuk menghitung jumlah status
    rekap = {"Normal": 0, "Warning": 0, "Critical": 0}
    
    for item in data:
        status = item["status"]
        if status in rekap:
            rekap[status] += 1
            
    return rekap
